scripts/OverlaySurvey.py

```diff
@@ -113,53 +154,40 @@ def update_results(graph, parent_info, parent_key, results, is_inbound):
         else:
             graph.add_edge(parent_key, other_key, **edge_properties)
 
-    if "numTotalInboundPeers" in parent_info:
-        results["totalInbound"] = parent_info["numTotalInboundPeers"]
-        graph.add_node(parent_key,
-                       numTotalInboundPeers=parent_info[
-                           "numTotalInboundPeers"
-                       ])
-
-    if "numTotalOutboundPeers" in parent_info:
-        results["totalOutbound"] = parent_info["numTotalOutboundPeers"]
-        graph.add_node(parent_key,
-                       numTotalOutboundPeers=parent_info[
-                           "numTotalOutboundPeers"
-                       ])
-
-    if "maxInboundPeerCount" in parent_info:
-        results["maxInboundPeerCount"] = parent_info["maxInboundPeerCount"]
-        graph.add_node(parent_key,
-                       maxInboundPeerCount=parent_info[
-                           "maxInboundPeerCount"
-                       ])
-
-    if "maxOutboundPeerCount" in parent_info:
-        results["maxOutboundPeerCount"] = parent_info["maxOutboundPeerCount"]
-        graph.add_node(parent_key,
-                       maxOutboundPeerCount=parent_info[
-                           "maxOutboundPeerCount"
-                       ])
-
-
-request_count = 0
-
-
-def send_requests(peer_list, params, request_url):
-    print("Requesting %s for %s peers" % (request_url, len(peer_list)))
-    limit = 10
-    # Submit `limit` queries roughly every ledger
-    for key in peer_list:
-        params["node"] = key
-        get_request(url=request_url, params=params)
-        print("Send request to %s" % key)
-        global request_count
-        request_count += 1
-        if (request_count % limit) == 0:
-            print("Submitted %i queries, sleep for ~1 ledger" % limit)
-            time.sleep(5)
-
-    print("Done")
+    # Add survey results to parent node (if available)
+    field_names = [("numTotalInboundPeers", "totalInbound"),
+                   ("numTotalOutboundPeers", "totalOutbound"),
+                   ("maxInboundPeerCount", "maxInboundPeerCount"),
+                   ("maxOutboundPeerCount", "maxOutboundPeerCount"),
+                   ("addedAuthenticatedPeers", "addedAuthenticatedPeers"),
+                   ("droppedAuthenticatedPeers", "droppedAuthenticatedPeers"),
+                   ("p75SCPFirstToSelfLatencyMs", "p75SCPFirstToSelfLatencyMs"),
+                   ("p75SCPSelfToOtherLatencyMs", "p75SCPSelfToOtherLatencyMs"),
+                   ("lostSyncCount", "lostSyncCount"),
+                   ("isValidator", "isValidator")]
+    update_node(graph, parent_info, parent_key, results, field_names)
+
+
+def send_survey_requests(peer_list, url_base):
+    """
+    Request survey data from a list of peers. `url_base` is the root HTTP
+    endpoint to send requests to.
+    """
+    request_url = url_base + "/surveytopologytimesliced"
+    logger.info("Requesting survey data from %s peers", len(peer_list))
```

Comment:
No more request rate throttling? (previous version of this code limited number of requests per ledger to 10)

Reply:
I removed the request throttling because stellar-core has its own builtin request throttling (see [`SurveyManager::topOffRequests`](https://github.com/bboston7/stellar-core/blob/0d339f979cfbe63e9e23f0350765286734ba5170/src/overlay/SurveyManager.cpp#L1006)). Specifically, stellar-core sends 10 requests from its queue every 15 seconds.

Even with the old request rate throttling this queue would build up because the script would add 10 nodes every 5 seconds, but stellar-core would only send a batch of 10 every 15 seconds.

That being said, I'm happy to add back in request rate throttling as a extra safety measure if you think it's necessary. Thoughts?

Reply:
That makes sense, I'm just trying to recall why we added this throttling in the first place. I agree though that it's weird to have throttling in both places, and ideally core should properly deal with all the requests. Looks like this [PR](https://github.com/stellar/stellar-core/pull/3633) added throttling because survey on the core side used to expire too quickly, I think this was related to the old `-d` flag. This shouldn't be the case with the new-style survey, since our reporting phase is now long by default, right?

Reply:
Ah, that makes sense. Digging into it, I think without the throttling in the script there's a weird race condition that would prematurely end the survey on the stellar-core side immediately after processing a large batch of requests. Trickling in requests causes the expiration to be regularly extended and avoids the premature expiration.

The V2 survey removed the `duration` parameter and instead [uses the survey phase](https://github.com/bboston7/stellar-core/blob/bcf2e59ab1e714914c6127f28887a2c5518fa45a/src/overlay/SurveyManager.cpp#L1169) to determine when the survey is finished.

Reply:
Yeah, that makes sense. I like that in this version we move the decision to end survey to the script itself after some inactivity rather than try to end in core.